airflow/jobs/scheduler_job_runner.py

```diff
@@ -1814,7 +1814,7 @@ def _orphan_unreferenced_datasets(self, session: Session = NEW_SESSION) -> None:
         An orphaned dataset is no longer referenced in any DAG schedule parameters or task outlets.
         """
         orphaned_dataset_query = session.scalars(
-            select(DatasetModel)
+            select(DatasetModel.id)
```

Comment:
I don’t think this is right? We use `scalars` here so `orphaned_dataset_query` would be a list of ints (ids) but we use the results as dataset objects below.

Reply:
Maybe we should also turn this into an `UPDATE` if possible